mlflow/utils/credentials.py

```diff
@@ -41,3 +46,129 @@ def read_mlflow_creds() -> MlflowCreds:
+        return True
+    except Exception:
+        return False
+
+
+def _overwrite_or_create_databricks_profile(
+    file_name,
+    profile,
+    profile_name="DEFAULT",
+):
+    """Overwrite or create a profile in the databricks config file.
+
+    Args:
+        file_name: string, the file name of the databricks config file, usually `~/.databrickscfg`.
+        profile: dict, contains the authentiacation profile information.
+        profile_name: string, the name of the profile to be overwritten or created.
+    """
+    profile_name = f"[{profile_name}]"
+    with open(file_name) as file:
+        lines = file.readlines()
+    start_index = -1
+    end_index = -1
+    # Find the start and end indices of the profile to overwrite.
+    for i in range(len(lines)):
+        if lines[i].strip() == profile_name:
+            start_index = i
+            break
+
+    if start_index != -1:
+        for i in range(start_index + 1, len(lines)):
+            # Reach an empty line or a new profile.
+            if lines[i].strip() == "" or lines[i].startswith("["):
+                end_index = i
+                break
+        end_index = end_index if end_index != -1 else len(lines)
+        del lines[start_index : end_index + 1]
+
+    # Write the new profile to the end of the file.
+    lines.append(profile_name + "\n")
+    lines.append(f"host = {profile['host']}\n")
+    if "token" in profile:
+        lines.append(f"token = {profile['token']}\n")
+    else:
+        lines.append(f"username = {profile['username']}\n")
+        lines.append(f"password = {profile['password']}\n")
+
+    # Write back the modified lines to the file.
+    with open(file_name, "w") as file:
+        file.writelines(lines)
+
+
+def _databricks_login():
+    """Set up databricks authentication and connect MLflow to Databricks tracking server."""
+    if _check_databricks_auth():
+        # Check if the auth has already been set.
+        _logger.info("Succesfully signed in Databricks!")
+        return
+
+    while True:
+        host = getpass.getpass("Databricks Host (should begin with https://): ")
+        if host.startswith("https://"):
+            break
+        _logger.error("Error: The host does not start with https://")
+
+    profile = {"host": host}
+    if "community" in host:
+        # Databricks community edition requires username and password for authentication.
+        username = getpass.getpass("Username: ")
+        password = getpass.getpass("Password: ")
+        profile["username"] = username
+        profile["password"] = password
+    else:
+        # Production or staging Databricks requires personal token for authentication.
+        token = getpass.getpass("Token: ")
+        profile["token"] = token
+
+    home = os.path.expanduser("~")
+    file_name = f"{home}/.databrickscfg"
+    profile_name = os.environ.get("DATABRICKS_CONFIG_PROFILE", "DEFAULT")
+    _overwrite_or_create_databricks_profile(file_name, profile, profile_name)
```

Comment:
Does the following code work on notebook environment?

```python
import subprocess

subprocess.Popen(["databricks", "configure", "--host", "https://test.cloud.databricks.com"], stdin=subprocess.PIPE).communicate(input=b"token\n")

```

Reply:
Does not work on Google colab